var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -142,3 +154,122 @@ def install(self, spec, prefix):
+                             join_path(self.prefix, 'lib', 'mic'))
+        run_env.prepend_path('MIC_LIBRARY_PATH',
+                             join_path(self.prefix, 'tbb', 'lib', 'mic'))
+        run_env.prepend_path('MIC_LD_LIBRARY_PATH',
+                             join_path(self.prefix, 'tbb', 'lib', 'mic'))
+
+        if self.spec.satisfies('+all'):
+            run_env.prepend_path('PATH',
+                                 join_path(self.prefix,
+                                           'debugger_{0}'.format(major_ver),
+                                           'gdb', 'intel64_mic', 'bin'))
+            run_env.prepend_path('LD_LIBRARY_PATH',
+                                 join_path(self.prefix,
+                                           'debugger_{0}'.format(major_ver),
+                                           'libipt', 'intel64', 'lib'))
+            run_env.set('GDBSERVER_MIC',
+                        join_path(self.prefix,
+                                  'debugger_{0}'.format(major_ver), 'gdb',
+                                  'targets', 'mic', 'bin', 'gdbserver'))
+            run_env.set('GDB_CROSS',
+                        join_path(self.prefix,
+                                  'debugger_{0}'.format(major_ver),
+                                  'gdb', 'intel64_mic', 'bin', 'gdb-mic'))
+            run_env.set('MPM_LAUNCHER',
+                        join_path(self.prefix,
+                                  'debugger_{0}'.format(major_ver), 'mpm',
+                                  'mic',
+                                  'bin', 'start_mpm.sh'))
+            run_env.set('INTEL_PYTHONHOME',
+                        join_path(self.prefix,
+                                  'debugger_{0}'.format(major_ver), 'python',
+                                  'intel64'))
+
+        if (self.spec.satisfies('+all') or self.spec.satisfies('+mpi')) and \
+           self.spec.satisfies('@cluster'):
+            run_env.prepend_path('PATH',
+                                 join_path(self.prefix, 'mpi', 'intel64',
+                                           'bin'))
+            run_env.prepend_path('LD_LIBRARY_PATH',
+                                 join_path(self.prefix, 'mpi', 'intel64',
+                                           'lib'))
+            run_env.prepend_path('LIBRARY_PATH',
+                                 join_path(self.prefix, 'mpi', 'intel64',
+                                           'lib'))
+            run_env.prepend_path('LD_LIBRARY_PATH',
+                                 join_path(self.prefix, 'mpi', 'mic', 'lib'))
+            run_env.prepend_path('MIC_LIBRARY_PATH',
+                                 join_path(self.prefix, 'mpi', 'mic', 'lib'))
+            run_env.prepend_path('MIC_LD_LIBRARY_PATH',
+                                 join_path(self.prefix, 'mpi', 'mic', 'lib'))
+            run_env.set('I_MPI_ROOT', join_path(self.prefix, 'mpi'))
```

Comment:
My Installation's prefix doesn't have an "mpi" sub directory. There is an impi/version directory though:

[lee218@rzgenie2:spack]$ ls opt/spack/linux-redhat7-x86_64/gcc-4.8.5/intel-parallel-studio-cluster.2016.3-sdj5atoyj6y5x7sq7t5plec2ut56lxsw/
Licenses                            inspector_xe_2016
advisor_xe                          inspector_xe_2016.1.3.460803
advisor_xe_2016                     ipp
advisor_xe_2016.1.40.463413         itac
bin                                 itac_9.1
clck                                itac_latest
clck_latest                         lib
compilers_and_libraries             man
compilers_and_libraries_2016        mkl
compilers_and_libraries_2016.3.210  mpirt
daal                                parallel_studio_xe_2016.3.067
debugger_2016                       samples_2016
documentation_2016                  tbb
ide_support_2016                    third-party-programs_ja.txt
imb                                 vtune_amplifier_xe
impi                                vtune_amplifier_xe_2016
include                             vtune_amplifier_xe_2016.3.0.463186
inspector_xe
[lee218@rzgenie2:spack]$ ls opt/spack/linux-redhat7-x86_64/gcc-4.8.5/intel-parallel-studio-cluster.2016.3-sdj5atoyj6y5x7sq7t5plec2ut56lxsw/impi/5.1.3.210/
benchmarks  binding  include64  lib64  mic
bin64       etc64    intel64    man    test